### ISSUE TITLE
Creating a wildcard hostname alias

### ISSUE THREAD

**Issue by ai-fwd** [Orckestra/C1-CMS-Foundation#481]
Looking at HostnameBindingsFacade.cs it doesn't seem possible to create a wildcard subdomain hostname (i.e. *.mydomain.com). Can someone confirm if this is/isn't supported since the documentation doesn't explicitly call it out.

**Comment by burningice2866** [comment]
In such case, isn't it easiest just to not have a Url Configuration? Or is it because you're having multiple sites each with their own wildcard hostname mapping?

**Comment by ai-fwd** [comment]
@burningice2866 not sure i understand the thinking behind not having a Url Configuration -- can you elaborate?

Yes, for my scenario i would have multiple sites each with their own IHostnameBinding (i.e. root and www -- mydomain.com & www.mydomain.com) and then in some cases i would need to have a separate IHostnameBinding that is the wildcard.

So, a concrete example would be:

mydomain.com, www.mydomain.com -> Website 1
*.mydomain.com -> Website 2

all on the same C1 instance.

**Comment by burningice2866** [comment]
In that case its a valid request, hence my question whether you have multiple websites.

In case you only have _***.mydomain.com -> Website**_ there is no need really for the Url Configuration

**Comment by burningice2866** [comment]
But if we take the feature request a step further and begin to think about the scenarios i quickly think of one problem;

If you want to render a absolute url to a page on your Website 2, what will the hostname part be then?

**Comment by ai-fwd** [comment]
@burningice2866 well i guess in that case you could use something like static.mydomain.com -- because it would pickup the subdomain and use the same routing logic just like the other subdomains -- it'd be up to the user to figure out if it made sense.

**Comment by burningice2866** [comment]
@ai-fwd "You" in this case refers to if C1 needs to render a absolute url to a page on Website 2. C1 wouldn't have a chance to know what actual hostname to use.

**Comment by mawtex** [comment]
@ai-fwd a solution could be to use www.mydomain.com as the primary host name and then put mydomain.com as an alias. C1 will redirect mydomain.com to www.mydomain.com. You get the desired effect, and another plus is on SEO, at least you should be immune to content dilution, meaning you will have unique URLs for pages and the self-proclaimed SEO experts of the world say [this is a good thing](https://raventools.com/blog/htaccess-redirects-to-a-single-hostname/).

In other words - we have support for multiple host names, but we force you to select a single host name to represent the content, because we're told that is smart SEO.

**Comment by ai-fwd** [comment]
@mawtex agree with it being an SEO best practice to have content tied to a single hostname but that would be in the scenario where the wildcard subdomains are just static alias.

In my scenario the content would actually change based on the subdomain. So s1.mydomain.com and s2.mydomain.com would actually show different content within the same C1 Website. 

I need to respect the URL and not have it redirected or changed. 

The bigger concern i have is around what @burningice2866 said on C1 wanting to show an absolute url -- do we know if this is something C1 does and where it does it?

**Comment by burningice2866** [comment]
@ai-fwd my suggestion for you would to contain this dynamic-content-by-hostname in a seperate website without a Url Configuration. 

I suppose there is no technical reason for you having to have both Website 1 and Website 2 in the same installation except for the ease of only having to manage one C1 installation instead of two?

**Comment by ai-fwd** [comment]
@burningice2866 that and the licensing. It seems a little heavy handed to host a new C1 instance (and infrastructure) for the scenario

**Comment by ai-fwd** [comment]
@burningice2866 @mawtex ok so based on your feedback i've implemented something that works for me. It might make sense to add this into the product but it's a little "magic" right now which likely needs to change. 

Basically i created a special hostname binding called "wildcard.mydomain.com" -- whenever a IHostnameBinding lookup occurs we do a secondary check to see if the incoming url would match the special binding. If it does, then we return the special binding.

It respects existing hostnames and alias and then only performs the subdomain/wildcard match if that fails.

I've done some light testing (all i have time for) and it seems to work well with no side affects so far for my scenario. 

Thanks for the help!
